src/category_theory/limits/shapes/types.lean

```diff
@@ -63,65 +63,98 @@ def initial_limit_cone : limits.colimit_cocone (functor.empty (Type u)) :=
 
 open category_theory.limits.walking_pair
 
-local attribute [tidy] tactic.case_bash
+/-- The product type `X × Y` forms a cone for the binary product of `X` and `Y`. -/
+-- We manually generate the other projection lemmas since the simp-normal form for the legs is
+-- otherwise not created correctly.
```

Comment:
What are they created as?

Reply:
The generated lemma for the legs is:
```
binary_product_cone_π_app :
  ∀ (X Y : Type u_1) (j : discrete walking_pair),
    (binary_product_cone X Y).π.app j = walking_pair.cases_on j prod.fst prod.snd
```
which is undesirable since the cases that are needed in practice are about `(binary_product_cone X Y).fst` and `.snd`, so for instance the implicit `tidy`s in `binary_product_functor` fail.